File: Functions/updateBudget.c
#include <stdio.h>
#include <math.h>
float income,rent,utilities,groceries,transportation, total_expenses, savings, total;

float input(char type[], float var){
    printf("how much is your %s: \n",type);
    scanf("%f", &var);
    return var;
}

void percent(char type[],int amount){
   int per = amount / income * 100;
   printf("Your %s is %d%% of your income.", type, per);
}

int main(void) {
    float prent,putilities,ptransportation, pgroceries,pexpenses;
    printf("this is going to calculate your budget for the month\n");
    income = input("income",income);
    rent = input("rent",rent);
    utilities = input("utilities",utilities);
    groceries = input("groceries",groceries);
    transportation = input("transportation",transportation);

    total_expenses = rent + utilities +groceries + transportation;
    savings = income *.2;
    total = income - total_expenses - savings;
    percent("rent", rent);
    
    percent("utilities", utilities);
    percent("groceries", groceries);
    percent("transportation", transportation);
    percent("expenses", total_expenses);

    printf("your income is:$%.2f\n", income);
    printf("your expenses are:$%.2f\n", total_expenses);
    printf("your savings are:$%.2f\n", savings);
    printf("your total leaft to spend is:$%.2f\n", total);
    return 0;
}

File: Functions/UpdateBudget.py
#Insert a function that asks for information
def ask(type):
    return float(input("How much is " +type+ "?: \n"))

#Insert a function that does pecents
def per(var):
    return float(var / income * 100)

#Gathering information
print("this is going to calculate your income!\n")
income = ask("income")
rent = ask("rent")
utilities = ask("utilities")
groceries = ask("groceries")
transportation = ask("transportation")
#using the info
expenses = rent + utilities + groceries + transportation
savings = income * .2
afterExpenses = income - expenses - savings
prent = per(rent)
putilities = per(utilities)
pgroceries = per(groceries)
ptransportation = per(transportation)

#Printing information
print("you spend " ,expenses)
print("you make " ,afterExpenses)
print("rent is " ,prent, "% of your total budget")
print("utilitites is " ,putilities, "% of your total budget")
print("groceries are " ,pgroceries, "% of your total budget")
print("transportation is " ,ptransportation, "% of your total budget")

